index.src.html

```diff
@@ -760,10 +760,10 @@ <h4 id="interfaces-credential-types-passwordcredential"><code>PasswordCredential
   :   <dfn attribute for="PasswordCredential">additionalData</dfn>
   ::  If the developer wishes to specify additional data to insert into the
       request body when submitting the credential information to a remote
-      endpoint, they can do so by assigning a {{FormData}} or
-      {{URLSearchParams}} object to this attribute. The credential information
-      will be mixed into the object to produce the body. The value is `null`
-      unless otherwise specified.
+      endpoint, they can do so by assigning a {{FormData}},
+      {{URLSearchParams}} or a generic object to this attribute. The
```

Comment:
> `record<>` doesn't show up in any Blink or Gecko IDL files. Does any UA implement it?

Yes. It's fairly new. @jyasskin added it for Bluetooth support.
> I'm trying to figure out how a record<DOMString, any> looks to a developer... That is, I think the bindings you've pointed to mean that something like: { "a": "b", "c": "d" } would be converted into « ("a", "b"), ("c", "d") ». Is that accurate?

Yes. It's basically an open-dictionary.

> The any as the value type means that we'd be back in the same boat with regard to nested structures' arbitrary code, right?

Indeed.

> Making record<> handle nesting seems like it would be complicated.

I'm looking at this right now for JSON serialization and I agree, it's complicated.

> Just to back up a bit so you're up to speed with what we're trying to do here: the goal is to give developers the ability to layer the credential management API on top of an existing form submission so that they can use it's magicalness to make users' lives easier. But: we'd like to do so without revealing the password data to JavaScript.
> 
> We currently rely on FormData and URLSearchParams to handle the basic cases of taking an existing form and submitting it's data (the developer gives us one or the other, we inject password data during the serialization process), but neither handles some more convoluted flows where developers submit data as JSON via XHR in order to sign a user in. We'd like to support those as well.
> 
> The idea at the moment is that a developer could build up the object structure they want to send to the server, and tell us what the field names ought to be for the username and password. We'll take the object they give us, set those two properties, and then serialize the whole thing.
> 
> I'd like to ensure that the act of serialization doesn't give malicious code a chance to walk the object structure and exfiltrate its data. Hence my worry about getters and setters and proxies and etc.

OK, that makes sense.

I think what you want is something similar to [serializable types](https://heycam.github.io/webidl/#dfn-serializable-type). Note these are missing records right now (see https://github.com/heycam/webidl/issues/262) and are being revamped in https://github.com/heycam/webidl/issues/188).